Assets/Stack mini game/Scripts/CubeMover.cs

```diff
@@ -20,13 +31,29 @@ private void Awake()
         }
     }
 
+    private void Update()
+    {
+        MoveCube();
+    }
+
     public void ChangeDirection()
     {
         direction *= -1;
     }
 
-    public void MoveCube()
+    private void MoveCube()
+    {
+        if(currentMovingCube != null && canMoveCube == true)
+            currentMovingCube.position += new Vector3(movingSpeed * 5 * Time.deltaTime, 0f, 0f) * direction;
+    }
+
+    private void ToggleMoveCube(bool onOff)
+    {
+        canMoveCube = onOff;
+    }
+
+    private void SetMovingCube(GameObject cube)

```

Comment:
I guess the reason I did that is because SetMovingCube subcribes to an event that has a Gameobject as a parameter. I have other methods subscribing to that event, that's why I used Gameobject as the type. I guess I could make different events with different argument types.